lldb/include/lldb/Symbol/SymbolFile.h

```diff
@@ -445,7 +445,11 @@ class SymbolFile : public PluginInterface {
   ///     contains the keys "type", "symfile", and "separate-debug-info-files".
   ///     "type" can be used to assume the structure of each object in
   ///     "separate-debug-info-files".
-  virtual bool GetSeparateDebugInfo(StructuredData::Dictionary &d) {
+  /// \param errors_only
+  ///     If true, then only return separate debug info files that encountered
+  ///     errors during loading.
+  virtual bool GetSeparateDebugInfo(StructuredData::Dictionary &d,
+                                    bool errors_only) {
```

Comment:
Not sure I'm a fan of a parameter that changes behavior. Why not introduce a new function entirely?

Reply:
> Not sure I'm a fan of a parameter that changes behavior. Why not introduce a new function entirely?

Hmm... I was thinking it would save some lines of boilerplate and avoid adding a new function to `SymbolFile`'s API. Also, it seems to me that there are other functions in `SymbolFile` that have a similar pattern?
```
virtual void FindFunctions(const Module::LookupInfo &lookup_info,
                             const CompilerDeclContext &parent_decl_ctx,
                             bool include_inlines, SymbolContextList &sc_list);
...
  virtual CompilerDeclContext
  FindNamespace(ConstString name, const CompilerDeclContext &parent_decl_ctx,
                bool only_root_namespaces = false) {
    return CompilerDeclContext();
  }
...
```

Reply:
I see. Does `GetSeparateDebugInfo` return all the successful entries and all the errors if `errors_only` is off? I guess what I'm asking is, is the flag there to filter or change the behavior entirely?

Reply:
Okay, so it doesn't do something completely different! That's what I wanted to avoid. No issue on my end then, thanks for clarifying.